Sources/WendyAgent/Services/ConnMan.swift

```diff
@@ -0,0 +1,474 @@
+                continue
+            }
+
+            // Merge array of dictionaries into a single dictionary
+            var properties: [DBusValue: DBusValue] = [:]
+            for propItem in propertiesArray {
+                if case .dictionary(let dict) = propItem {
+                    for (key, value) in dict {
+                        properties[key] = value
+                    }
+                }
+            }
+
+            services.append((path: path, properties: properties))
+        }
+
+        return services
+    }
+
+    /// Extract SSID from service properties
+    private func extractSSID(from properties: [DBusValue: DBusValue]) -> String? {
+        guard let nameValue = properties[DBusValue.string("Name")],
+            case .variant(let variant) = nameValue,
+            case .string(let name) = variant.value
+        else {
+            return nil
+        }
+        return name
+    }
+
+    /// Extract signal strength from service properties
+    private func extractSignalStrength(from properties: [DBusValue: DBusValue]) -> Int8? {
+        guard let strengthValue = properties[DBusValue.string("Strength")],
+            case .variant(let variant) = strengthValue,
+            case .byte(let strength) = variant.value
+        else {
+            return nil
+        }
+        return Int8(strength)
+    }
+
+    /// Extract security type from service properties
+    private func extractSecurity(from properties: [DBusValue: DBusValue]) -> Bool {
+        guard let securityValue = properties[DBusValue.string("Security")],
+            case .variant(let variant) = securityValue,
+            let array = variant.value.array,
+            !array.isEmpty
+        else {
+            return false
+        }
+
+        for item in array {
+            if case .string(let security) = item,
+                security != "none"
+            {
+                return true
+            }
+        }
+
+        return false
+    }
+
+    /// Extract service type from properties
+    private func extractServiceType(from properties: [DBusValue: DBusValue]) -> String? {
+        guard let typeValue = properties[DBusValue.string("Type")],
+            case .variant(let variant) = typeValue,
+            case .string(let type) = variant.value
+        else {
+            return nil
+        }
+        return type
+    }
+
+    /// Extract service state from properties
+    private func extractServiceState(from properties: [DBusValue: DBusValue]) -> String? {
+        guard let stateValue = properties[DBusValue.string("State")],
+            case .variant(let variant) = stateValue,
+            case .string(let state) = variant.value
+        else {
+            return nil
+        }
+        return state
+    }
+
+    /// Extract IP address from service properties
+    private func extractIPAddress(from properties: [DBusValue: DBusValue]) -> String? {
+        guard let ipv4Value = properties[DBusValue.string("IPv4")],
```

Comment:
worth calling out in the doc part and in the comments for this struct that it supports only IPv4